app/src/main/java/de/westnordost/streetcomplete/osm/address/HouseNumbers.kt

```diff
@@ -89,6 +89,14 @@ data class HouseNumberWithLetter(
 ) : StructuredHouseNumber() {
     override fun toString() = "$number$separator$letter"
 }
+/** e.g. C-12 */
+data class HouseNumberWithLetterAtStart(
```

Comment:
1. C-12a, C-12/3 not possible with this data structure. But I don't see a reason why not. In the end, the letter at the beginning is more or less something like a `addr:block_number` baked into the `addr:housenumber`.

2. I think the `compareTo` method in `StructuredHouseNumber` should be adapted too?

3. Given 1, maybe it would make sense first to inquire why `addr:block_number` is not used in places where `C-13` is used as housenumber

Reply:
> 3. Given 1, maybe it would make sense first to inquire why `addr:block_number` is not used in places where `C-13` is used as housenumber

What is clear is that it is not used (`addr:block_number` is used very rarely, and not with a letter).
@letypequividelespoubelles do you have some more information about this? Is the letter really actually a block number?